PrivateHospitals/backend/PrivateHospitals/PrivateHospitals.Application/Services/Appointment/AppointmentService.cs

```diff
@@ -0,0 +1,101 @@
+using AutoMapper;
+using PrivateHospitals.Application.Dtos.Appointment;
+using PrivateHospitals.Application.Dtos.Doctor;
+using PrivateHospitals.Application.Dtos.Patient;
+using PrivateHospitals.Application.Interfaces.Appointment;
+using PrivateHospitals.Application.Responses;
+using PrivateHospitals.Core.Enum;
+using PrivateHospitals.Core.Models.Users;
+using PrivateHospitals.Infrastructure.Interfaces.Appointment;
+using PrivateHospitals.Infrastructure.Interfaces.Doctor;
+using PrivateHospitals.Infrastructure.Interfaces.Patient;
+
+namespace PrivateHospitals.Application.Services.Appointment;
+
+public class AppointmentService(
+    IAppointmentRepository _appointmentRepository,
+    IDoctorRepository _doctorRepository,
+    IMapper _mapper,
+    IPatientRepository _patientRepository
+): IAppointmentService
+{
+    public async Task<Result<bool>> CreateAppointment(CreateAppointmentDto appointmentDto)
+    {
+        var doctorDto = await _doctorRepository
+            .GetDoctorByFullName(appointmentDto.DoctorFirstName, appointmentDto.DoctorLastName);
```

Comment:
Let's use doctor id here. It will require index for doctor names but this index looks redundant. Let's not load our DB when not needed